### PR TITLE
Updates release-manifests Target

### PR DESCRIPTION
**Pull request by danehans**
Updates the `release-manifests` target to include Infra Manager RBAC.

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

**Comment by arkodg** [comment]
@danehans we could reuse a lot of the existing logic . This diff seems to work with `IMAGE=envoyproxy/gateway TAG=foo make release-manifests`
```
git diff
diff --git a/tools/make/kube.mk b/tools/make/kube.mk
index c879c88..6721e8a 100644
--- a/tools/make/kube.mk
+++ b/tools/make/kube.mk
@@ -34,12 +34,16 @@ ifndef ignore-not-found
   ignore-not-found = false
 endif
 
-.PHONY: kube-install
-kube-install: manifests $(tools/kustomize) ## Install Envoy Gateway CRDs into the Kubernetes cluster specified in ~/.kube/config.
+.PHONY: copy-manifests
+copy-manifests:
        mkdir -pv $(OUTPUT_DIR)/manifests/provider
        cp -r $(KUBE_PROVIDER_DIR) $(OUTPUT_DIR)/manifests/provider
        mkdir -pv $(OUTPUT_DIR)/manifests/infra
        cp -r $(KUBE_INFRA_DIR) $(OUTPUT_DIR)/manifests/infra
+       cd $(OUTPUT_DIR)/manifests/provider/config/envoy-gateway && $(ROOT_DIR)/$(tools/kustomize) edit set image envoyproxy/gateway-dev=$(IMAGE):$(TAG)
+
+.PHONY: kube-install
+kube-install: manifests $(tools/kustomize) copy-manifests ## Install Envoy Gateway CRDs into the Kubernetes cluster specified in ~/.kube/config.
        $(tools/kustomize) build $(OUTPUT_DIR)/manifests/provider/config/crd | kubectl apply -f -
        kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/${GATEWAY_API_VERSION}/experimental-install.yaml
 
@@ -49,7 +53,6 @@ kube-uninstall: manifests $(tools/kustomize) ## Uninstall Envoy Gateway CRDs fro
 
 .PHONY: kube-deploy
 kube-deploy: kube-install ## Install Envoy Gateway controller into the Kubernetes cluster specified in ~/.kube/config.
-       cd $(OUTPUT_DIR)/manifests/provider/config/envoy-gateway && $(ROOT_DIR)/$(tools/kustomize) edit set image envoyproxy/gateway-dev=$(IMAGE):$(TAG)
        $(tools/kustomize) build $(OUTPUT_DIR)/manifests/provider/config/default | kubectl apply -f -
        $(tools/kustomize) build $(OUTPUT_DIR)/manifests/infra/config/rbac | kubectl apply -f -
 
@@ -104,5 +107,7 @@ delete-cluster: $(tools/kind) ## Delete kind cluster.
        $(tools/kind) delete cluster --name envoy-gateway
 
 .PHONY: release-manifests
-release-manifests: $(tools/kustomize) ## Generate Kubernetes release manifests.
-       tools/hack/release-manifests.sh $(GATEWAY_API_VERSION) $(TAG)
+release-manifests: $(tools/kustomize) copy-manifests ## Generate Kubernetes release manifests.
+       mkdir -p $(ROOT_DIR)/release-artifacts
+       curl -sLo $(ROOT_DIR)/release-artifacts/gatewayapi-crds.yaml https://github.com/kubernetes-sigs/gateway-api/releases/download/${GATEWAY_API_VERSION}/experimental-install.yaml
+       $(tools/kustomize) build $(OUTPUT_DIR)/manifests/provider/config/default > $(ROOT_DIR)/release-artifacts/install.yaml
```

File: kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+resources:
```
Comment (arkodg) [inline kustomization.yaml:1]: with this change we'll have kustomization files living in 3 places - `root`, `internal/provider/kubernetes/` and `internal/infrastructure/kubernetes`, can we simplify this ?

Comment (danehans) [reply kustomization.yaml:1]: I'm concerned about making too many changes during the RC2 release. I have confirmed that this PR works so I'd like to proceed with it and finish the release. I created https://github.com/envoyproxy/gateway/issues/479 to discuss how to improve the current kustomization setup.

**Comment by danehans** [comment]
Regarding https://github.com/envoyproxy/gateway/pull/478#issuecomment-1263895212, this suggestion does not work. The install manifest does not include the infra manager [RBAC](https://github.com/envoyproxy/gateway/tree/main/internal/infrastructure/kubernetes/config/rbac).

**Review by arkodg — APPROVED** [review]

**Review by skriss — APPROVED** [review]

**Pull request merged**